src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs

```diff
@@ -489,10 +489,13 @@ private void DeriveTrxFilePath()
             {
                 if (!string.IsNullOrWhiteSpace(logFilePrefixValue))              
                 {
-                    var framework = this.parametersDictionary[DefaultLoggerParameterNames.TargetFramework] ?? string.Empty;
-                    framework = NuGetFramework.Parse(framework).GetShortFolderName();
+                    var framework = this.parametersDictionary[DefaultLoggerParameterNames.TargetFramework];
+                    if (framework != null)
+                    {
+                        framework = NuGetFramework.Parse(framework).GetShortFolderName();
+                    }
 
-                    logFilePrefixValue = logFilePrefixValue.Replace(".trx", string.Empty) + "_" + framework + DateTime.Now.ToString("_yyyyMMddHHmmss", DateTimeFormatInfo.InvariantInfo) + this.trxFileExtension;
+                    logFilePrefixValue = logFilePrefixValue + "_" + framework + DateTime.Now.ToString("_yyyyMMddHHmmss", DateTimeFormatInfo.InvariantInfo) + this.trxFileExtension;
```

Comment:
logFilePrefixValue = logFilePrefixValue + "_" + framework only when framework value is not null